src/app/components/hris/employees/employee-profile/accordions/accordion-banking/accordion-banking.component.ts

```diff
@@ -164,56 +164,58 @@ export class AccordionBankingComponent {
-          this.getEmployeeBankingData();
-          this.checkBankingInformationProgress();
-          this.totalBankingProgress();
-          this.employeeBankingStarterkitService.getAllBankingAndStarterkits();
-          this.hasUpdatedBanking = true;
-          this.editBanking = false;
-          this.employeeBankingsForm.disable();
-        },
-        error: (error) => {
-          this.snackBarService.showSnackbar("Unable to Save Banking Information", "snack-error");
-        }
-      })
+    if(this.bankingPDFName.length >= 1){
+      this.editBanking = false;
+      this.isUpdated = true;
+      const employeeBankingFormValue = this.employeeBankingsForm.value;
+      this.employeeBankingDto = {
+        id: this.bankingId,
+        employeeId: this.employeeProfile?.id,
+        bankName: employeeBankingFormValue.bankName,
+        branch: `${employeeBankingFormValue.branch}`,
+        accountNo: `${employeeBankingFormValue.accountNo}`,
+        accountType: employeeBankingFormValue.accountType,
+        status: 1,
+        declineReason: this.bankingReason,
+        file: employeeBankingFormValue.file
+      }
+      if (this.hasBankingData) {
+        this.employeeBankingService.updatePending(this.employeeBankingDto).subscribe({
+          next: () => {
+            this.addOrUpdateBanking("Updated")
+          },
+          error: (error) => {
+            this.snackBarService.showSnackbar("Unable to Save Banking Information", "snack-error");
+          }
+        })
+      }
+      else {
+        this.employeeBankingService.addBankingDetails(this.employeeBankingDto).subscribe({
+          next: () => {
+            this.addOrUpdateBanking("Saved")
+          }
+          , error: (error) => {
+            this.snackBarService.showSnackbar("Unable to Save Banking Information", "snack-error");
+          }
+        })
+      }
     }
-    else {
-      this.employeeBankingService.addBankingDetails(this.employeeBankingDto).subscribe({
-        next: () => {
-          this.snackBarService.showSnackbar("Saved", "snack-success");
-          this.getEmployeeBankingData();
-          this.checkBankingInformationProgress();
-          this.totalBankingProgress();
-          this.employeeBankingStarterkitService.incrementPendingCount();
-          this.hasUpdatedBanking = true;
-          this.editBanking = false;
-          this.employeeBankingsForm.disable();
-        }
-        , error: (error) => {
-          this.snackBarService.showSnackbar("Unable to Save Banking Information", "snack-error");
-        }
-      })
+    else{
+      this.snackBarService.showSnackbar("Add a Proof of account", "snack-error")
     }
   }
 
+  addOrUpdateBanking(message: string) {
+    this.snackBarService.showSnackbar(message, "snack-success");
+    this.getEmployeeBankingData();
+    this.checkBankingInformationProgress();
+    this.totalBankingProgress();
+    this.employeeBankingStarterkitService.getAllBankingAndStarterkits();
```

Comment:
For updatePending
```suggestion
    this.employeeBankingStarterkitService.getAllBankingAndStarterkits();
```
For addBankingDetails
```suggestion
    this.employeeBankingStarterkitService.incrementPendingCount();
```